### DISCUSSION TITLE
Embed code available ?

### DISCUSSION THREAD

**Discussion by EricGuic** [Hubro/clipface#28] [Q&A] [answered]
Hi,
I'm wondering if, alongside the link clipface already provide, I can get an embed code for the video ?
For example to be added in the post of a Ghost blog ?
Thanks.

PS : On the home page of the project, you put at the bottom a link to the Clipface v2 "roadmap" (or planned tasks, I don't remember the exact term), but it's broken and leads to a 404 page.

**Comment by Hubro** [comment, marked as answer]
> Hi,
> I'm wondering if, alongside the link clipface already provide, I can get an embed code for the video ?
> For example to be added in the post of a Ghost blog ?

Do you mean a HTML embed using an iframe? Does Ghost blog accept HTML?

> PS : On the home page of the project, you put at the bottom a link to the Clipface v2 "roadmap" (or planned tasks, I don't remember the exact term), but it's broken and leads to a 404 page.

Thanks, I deleted the "Project" page the other day, as I don't like that feature. I'll be using milestones instead. Thanks for reminding me about the REAME link, I'll fix it promptly.

**Reply by EricGuic** [reply]
Hi Hubro, thanks for your quick reply, and thanks for your work ;-)

Yes, I primarly need iframe code, because it's seems to be the most "universal" way to embed video in a large variety of CMS.
Until now, I was using Wordpress, but want to try Ghost to see if it's simplier (because Wordpress IMHO has to many features and lose it's original simplicity).
So I don't have to much experience of Ghost for now. You can't upload your video like you do with wordpress, and that you have to host your video somewhere else. Ghost offer a simple integration with Youtube or Vimeo (you just have to provide the URL of the video), but I want to self host my video.
So the iframe is the fisrt way to do it.

I've found some guide online saying that you can use a video tag like this (but I can't try it now) : 
`<video controls width="500">
    <source src="https://YOURSITE.COM/YOUR-FILE.MP4"
            type="video/mp4">
    Sorry, your browser doesn't support embedded videos.
</video>`

Thanks for your advice. Eric

**Reply by EricGuic** [reply]
By the way, in my install, I plan to deploy Clipface (as the Ghost blog) inside a docker container on my NAS. 
And I run my NAS alongside a raspberry pi which run Nginx Proxy Manager. 
So the url will be : 
- all 80 and 443 request are send to the raspberry pi with nginx proxy manager
- the NAS is behind : https://nas.mydomain.com (port 5001 only for administration)
- clipface will be behind https://video.mydomain.com (for example on port 8080 of the NAS)
- the Ghost blog will also run on a container of the NAS, behind : https://web.mydomain.com (for example on port 8081 of the NAS)

So, for each video clip I want to embed in Ghost, I imagine to get a "video tag syntax" or an "iframe embed code" with a path to the video file like : 
https://video.mydomain.com/fheubkeushkefhh 
right ?

**Reply by Hubro** [reply]
> I've found some guide online saying that you can use a video tag like this (but I can't try it now) :
> ```
> <video controls width="500"> <source src="https://YOURSITE.COM/YOUR-FILE.MP4" type="video/mp4"> Sorry, your browser doesn't support embedded videos. </video>
> ```

If so, then it should be as simple as right clicking the video in Clipface as selecting "Copy video address..." and pasting the result into the `src="..."` parameter. (If you have authentication enabled, make sure you add the `?token=...` query parameter to the URL.)

Let me know if that works. With Clipface, there currently isn't any benefit in using an iframe embed over just using a video tag, since Clipface doesn't have a custom user interface for the video player. It's just a plain video tag.

**Reply by EricGuic** [reply]
Hi Hubro, and thanks for your reply.
I will try to deploy the container and test it.
Thanks for your help.
Eric

**Reply by EricGuic** [reply]
I've tested this and yes, my Ghost blog accept the simple video tag. Thanks for the tip ;-)
